### PR TITLE
make check for toolchain value in dependency spec in easyconfigs test suite aware that dumped easyconfig uses SYSTEM constant

### PR DESCRIPTION
**Pull request by boegel**
requires:
* https://github.com/easybuilders/easybuild-framework/pull/4069

### CODE REVIEW

**Comment by boegel** [comment]
The tests for this PR will fail until https://github.com/easybuilders/easybuild-framework/pull/4069 is merged, but this PR will need to get merged ASAP after the merging of https://github.com/easybuilders/easybuild-framework/pull/4069 to avoid broken tests in PRs being opened...

**Review by Micket — APPROVED** [review]
lgtm

File: test/easyconfigs/easyconfigs.py [hunk 1]
```diff
@@ -1429,7 +1430,13 @@ def template_easyconfig_test(self, spec):
                 # 4th value is toolchain spec
                 if len(dumped_dep) >= 4:
                     if len(orig_dep) >= 4:
-                        self.assertEqual(dumped_dep[3], orig_dep[3])
+                        # if True was used to indicate that dependency should use system toolchain,
+                        # then we need to compare the value for the dumped easyconfig more carefully;
+                        # see also https://github.com/easybuilders/easybuild-framework/pull/4069
+                        if orig_dep[3] == True:
```
Comment (Micket) [inline test/easyconfigs/easyconfigs.py:1436]:
```suggestion
                        if orig_dep[3] is True:
```

**Review by Micket — CHANGES_REQUESTED** [review]

File: test/easyconfigs/easyconfigs.py (hunk 1 above)
Comment (Micket) [reply test/easyconfigs/easyconfigs.py:1436]: i guess this is the nicest in this setting, don't want implicit true conversions i think

**Review by jfgrimm — APPROVED** [review]

**Pull request merged**